### ISSUE TITLE
adding nan_euclidean metric

### ISSUE THREAD

**Issue by PolarBean** [facebookresearch/faiss#3355]
# Summary

Sklearn has recently added support for the nan_euclidean as detailed in this issue https://github.com/scikit-learn/scikit-learn/issues/25319

Is it possible this will be added to faiss in the future? thank you.

**Comment by mdouze** [comment]
We can make an enhancement of it. 
For new metrics, the first stage is to support them for brute force search in 

https://github.com/facebookresearch/faiss/blob/main/faiss/utils/extra_distances-inl.h

this will enable them for FlatCodes search and HNSW search.

**Comment by MarcYin** [comment]
Is this going to be implemented for the gpu case?
